roles/translator/src/lib/mod.rs

```diff
@@ -1,7 +1,204 @@
+        let (tx_sv1_notify, _rx_sv1_notify): (
+            broadcast::Sender<server_to_client::Notify>,
+            broadcast::Receiver<server_to_client::Notify>,
+        ) = broadcast::channel(10);
+        let diff_config = Arc::new(Mutex::new(config.upstream_difficulty_config.clone()));
+        let target = Arc::new(Mutex::new(vec![0; 32]));
+
+        // Instantiate a new `Upstream` (SV2 Pool)
+        let upstream = match upstream_sv2::Upstream::new(
+            self.upstream.clone(),
+            config.upstream_authority_pubkey,
+            rx_sv2_submit_shares_ext,
+            tx_sv2_set_new_prev_hash,
+            tx_sv2_new_ext_mining_job,
+            config.min_extranonce2_size,
+            tx_sv2_extranonce,
+            status::Sender::Upstream(tx_status.clone()),
+            target.clone(),
+            diff_config.clone(),
+        )
+        .await
+        {
+            Ok(upstream) => upstream,
+            Err(e) => {
+                error!("Failed to create upstream: {}", e);
+                panic!();
+            }
+        };
+
+        // Spawn a task to do all of this init work so that the main thread can listen for signals
+        // and failures on the status channel. This allows for the tproxy to fail gracefully if any
+        // of these init tasks fail
+        let downstream_difficulty_config = config.downstream_difficulty_config.clone();
+        task::spawn(async move {
+            // Connect to the SV2 Upstream role
+            match upstream_sv2::Upstream::connect(
+                upstream.clone(),
+                config.min_supported_version,
+                config.max_supported_version,
+            )
+            .await
+            {
+                Ok(_) => info!("Connected to Upstream!"),
+                Err(e) => {
+                    error!("Failed to connect to Upstream EXITING! : {}", e);
+                panic!();
+                }
+            }
+
+            // Start receiving messages from the SV2 Upstream role
+            if let Err(e) = upstream_sv2::Upstream::parse_incoming(upstream.clone()) {
+                error!("failed to create sv2 parser: {}", e);
+                panic!();
+            }
+
+            debug!("Finished starting upstream listener");
+            // Start task handler to receive submits from the SV1 Downstream role once it connects
+            if let Err(e) = upstream_sv2::Upstream::handle_submit(upstream.clone()) {
+                error!("Failed to create submit handler: {}", e);
+                return;
+            }
+
+            // Receive the extranonce information from the Upstream role to send to the Downstream role
+            // once it connects also used to initialize the bridge
+            let (extended_extranonce, up_id) = rx_sv2_extranonce.recv().await.unwrap();
+            loop {
+                let target: [u8; 32] = target.safe_lock(|t| t.clone()).unwrap().try_into().unwrap();
+                if target != [0; 32] {
```

Comment:
I assume this is becasue we expect an all 0 target until we get something from the pool?

Reply:
we should avoid using PR comments to try to understand general implementation details, unless they have something to do with the goal of the PR at hand

otherwise, we make the review process noisier than it needs to be

the code here was just copy-pasted from `main.rs`, and we're not really aiming to change any of this logic on this PR... so even though it's showing up on the diff here, there's not much for us to care about it

I don't want to discourage @marathon-gary curiosity so I created this Discussion #1144 so we can ask questions about implementation details of `roles`